daemon/devices.go

```diff
@@ -37,6 +38,19 @@ func registerDeviceDriver(name string, d *deviceDriver) {
 	deviceDrivers[name] = d
 }
 
+func registerDeviceDrivers(cfg *config.Config) *cdi.Cache {
+	registerNvidiaDriver()
+
+	// Register the CDI driver before the daemon starts, as it might try to restore containers that depend on the CDI driver.
+	// Note that CDI is not inherently linux-specific, there are some linux-specific assumptions / implementations in the code that
+	// queries the properties of device on the host as well as performs the injection of device nodes and their access permissions into the OCI spec.
+	//
+	// In order to lift this restriction the following would have to be addressed:
+	// - Support needs to be added to the cdi package for injecting Windows devices: https://tags.cncf.io/container-device-interface/issues/28
+	// - The DeviceRequests API must be extended to non-linux platforms.
+	return registerCDIDriver(cfg.CDISpecDirs...)
```

Comment:
We should still only register it on Linux, previously it was checked with: 
https://github.com/moby/moby/blob/9a9cade0dfa872fafa98ab15fe51ee152d13649d/cmd/dockerd/daemon.go#L1062-L1064

now the whole function is unused and could be deleted
https://github.com/moby/moby/blob/9a9cade0dfa872fafa98ab15fe51ee152d13649d/cmd/dockerd/daemon.go#L1061-L1070